sources/api/netdog/src/wicked.rs

```diff
@@ -0,0 +1,504 @@
+//! The wicked module contains the data structures and functions needed to create network interface
+//! configuration files for wicked.
+//!
+//! The structures in this module are meant to be created from the user-facing structures in the
+//! `net_config` module or from a string, hence the `From` and `FromStr` implementations.  `Default` implementations for
+//! WickedInterface exist here as well, which provide sensible defaults in the absence of a desired
+//! network configuration.
+//!
+//! It's also important to note that there are network configuration options here that are not
+//! exposed to the user via `NetConfig`.  These options exist only to support the current sane
+//! defaults in place for the AWS platform.
+use crate::{
+    interface_name::InterfaceName,
+    net_config::{Dhcp4Config, NetInterface},
+};
+use serde::Serialize;
+use snafu::{ensure, OptionExt, ResultExt};
+use std::{collections::HashSet, convert::TryInto, fs, ops::Deref, path::Path, str::FromStr};
```

Comment:
nit: I'm used to seeing each `use std::X` import on separate lines, as though each were a different crate.